Storage/OAuthStorage.php

```diff
@@ -163,6 +168,24 @@ public function checkUserCredentials(IOAuth2Client $client, $username, $password
             return false;
         }
 
+        if ($user instanceof AdvancedUserInterface) {
+            if (!$user->isAccountNonLocked()) {
+                throw new LockedException('Account is blocked');
+            }
+
+            if (!$user->isAccountNonExpired()) {
+                throw new AccountExpiredException('Account is expired');
+            }
+
+            if (!$user->isCredentialsNonExpired()) {
+                throw new CredentialsExpiredException('Credentials is expired');
+            }
+
+            if (!$user->isEnabled()) {
+                throw new DisabledException('Account is disabled');
+            }
+        }
```

Comment:
I think it's best to do these checks after the username/password has been checked. This makes it harder for clients to fish for existing usernames.